hera_cal/abscal.py

```diff
@@ -3693,6 +3693,232 @@ def post_redcal_abscal_run(data_file, redcal_file, model_files, raw_auto_file=No
+        overwrite outputs if True.
+    tol: float, optional
+        perform calibration loop until differences in gain solutions are at this level.
+    max_iter: int, optional
+        maximum number of iterations to perform
+        default is 10
+    refant: tuple, optional
+        referance antenna in form of (antnum, polstr).
+        Default is None -> automatically select a refant.
+    ant_threshold: float, optional
+        threshold of flags in frequency and time for a given antenna to completely
+        flag this antenna rom calibration.
+    no_ampcal: bool, optional
+        skip amplitude calibration (only calibrate phases)
+        default is False.
+    no_phscal: bool, optional
+        skip phase calibration (only calibrate amplitudes)
+        default is False.
+    dly_lincal: bool, optional
+        perform initial delay calibration before abscal
+        default is False.
+    spoof_missing_channels: bool, optional
+        insert flagged gains in any frequency discontinunities.
+        default is False.
+    verbose: bool, optional
+        lots of outputs.
+    """
+    hdd = io.to_HERAData(data_file, filetype="uvh5")
+    if hasattr(hdd, "data_array") and hdd.data_array is not None:
+        data, data_flags, data_nsamples = hdd.build_datacontainers()
+    else:
+        data, data_flags, data_nsamples = hdd.read()
+
+    hdm = io.to_HERAData(model_file, filetype="uvh5")
+    if not hasattr(hdm, "data_array") or hdm.data_array is None:
+        hdm.read()
+
+    if precalibration_gain_file is not None:
+        uvc_precal = io.HERACal(precalibration_gain_file)
+        gains_precal, flags_precal, _, _ = uvc_precal.read()
+        # apply precal gains to data
+        calibrate_in_place(data=data, data_flags=data_flags,
+                           new_gains=gains_precal, cal_flags=flags_precal)
+
+    # expand hdm by redundancy
+    if inflate_model_by_redundancy:
+        if constrain_model_to_data_ants:
+            # In order to avoid inflating to full interferometer dataset
+            # which will be unecessary for many earlier analyses
+            # prune model antennas to only include data antennas in hdd
+            # this will only work if the antennas in the baseline keys for each redundant group
+            # in the model are also present in the dataset so only use this if you are confident
+            # that this is the case.
+            all_data_ants = np.unique(np.hstack([hdd.ant_1_array, hdd.ant_2_array]))
+            all_model_ants = np.unique(np.hstack([hdm.ant_1_array, hdm.ant_2_array]))
+            assert np.all([ant in all_data_ants for ant in all_model_ants]), "All model antennas must be present in data if constrain_model_to_data_ants is True!"
+            hdm.select(antenna_nums=all_data_ants,
+                       keep_all_metadata=False)
+        hdm.inflate_by_redundancy()
+        # also make sure to only include baselines present in hdd.
+        hdm.select(bls=hdd.bls)
```

Comment:
I think you need to follow this select with hdm._determine_blpair_slice and hdm._determine_pol_slice

Reply:
Added 
```        
hdm._determine_blt_slicing()
hdm._determine_pol_indexing()
```